### ISSUE TITLE
Multiple wireguard peers configs with the same public key causes wrong peer to be used/written to config

### ISSUE THREAD

**Issue by syserr0r** [opnsense/core#7698]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Enabling/disabling wireguard peers appears to enable/disable the wrong peer when the two peer configs overlap on public key.

Site A has multiple WAN connections, including; WAN and WAN-VDSL
Site B has 2 site-to-site wireguard peers configured for Site A named `SiteA` and `SiteA-VDSL`.
The primary difference is the `Endpoint Address` (although there could be other differences).

Regardless of which one peer is enabled, or both, only `SiteA-VDSL` shows in the `Status` tab under `WireGuard`.

**To Reproduce**

Steps to reproduce the behaviour:
1. Add two wireguard peers with the same public key
2. Enable any one, or both, peers
3. Observe that only one peer is ever shown in the status tab, regardless of which peers were enabled

**Expected behavior**

The relevant peers should be enabled/disabled as expected.

**Describe alternatives you considered**

If unsupported then Ideally the frontend would allow multiple peers to be added for the same public key, but only one peer for each public key can be enabled (assuming that is what the peers are identified by) - failing that then outright preventing duplicate public keys.

I would note it is useful to have multiple (inactive) configs that can be easily switched between.

**Additional context**

This was observed after upgrading from 24.1 to 24.7 - I am not sure if this was an issue before (it did not appear to be but I didn't check thoroughly)

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 24.7_9 (amd64).

**Comment by syserr0r** [comment]
It would appear that whilst disabling the peer from the `Peers` tab does not work; removing the undesired peer from the `Instances` view's `Peers` list _does_ provide the expected behaviour - so this can be used as a work-around in the interim.

**Comment by AdSchellevis** [comment]
The key used to identify records is a combined set of interface (`wgX` + `puplic-key`)

https://github.com/opnsense/core/blob/9240c5a15cd3164cd745dbe4456fd0213f6eab0d/src/opnsense/mvc/app/controllers/OPNsense/Wireguard/Api/ServiceController.php#L94

If your about differs from what `wg show` has to offer on the console, there's something to look into, but I likely need some screenshots and data samples.

**Comment by syserr0r** [comment]
I haven't tried looking at the console output, but given `wgX` + `public-key` are the key then yes, it isn't sufficient to distinguish between multiple entries with the same key and therefore writes the wrong one to the config file.

I'll leave it up to you if this "working as intended" or if it could be improved, but it caught me out slightly.

**Comment by AdSchellevis** [comment]
not sure what you mean without output, but per interface a public key sounds rather unique indeed

**Comment by syserr0r** [comment]
I'm not sure how I managed it but I had 2 peers with the same public key (one was using a backup link - they had different endpoint addresses but pointed to the same host).

On trying to reproduce it appears the UI doesn't allow duplicate public keys; so I'm not sure how I managed that.

I had mistakenly thought you were supporting duplicate public keys and was therefore trying to improve how that was handled. I can now see that duplicate public keys are not supported.
